### PR TITLE
feat: add project for core22 + snapcraft 7

### PR DESCRIPTION
**Pull request by tigarmo**
The rock is missing a lot of what the original snapcraft snap has but this is a good starting point. This is able to build 'hello' snaps (including from chisel slices).

### CODE REVIEW

File: rockcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+  only.
+license: GPL-3.0
+platforms:
+  amd64: #TODO
+
+parts:
+  snapcraft:
+    plugin: python
+    source: https://github.com/snapcore/snapcraft.git
+    source-tag: 7.5.4
```
Comment (sergiusens) [inline rockcraft.yaml:20]: try $CRAFT_PROJECT_VERSION

File: rockcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+      - ca-certificates
+    organize:
+      bin/chisel: usr/libexec/snapcraft/chisel
+
+entrypoint-service: snapcraft
+
+services:
+  snapcraft:
+    override: replace
+    command: /bin/snapcraft [ pack ]
```
Comment (lengau) [inline rockcraft.yaml:106]: Should this use `--destructive-mode` by default?

Comment (tigarmo) [reply rockcraft.yaml:106]: the `SNAPCRAFT_BUILD_ENVIRONMENT: host` call below takes care of that. The command line parameter only apply to some subcommands so it's hard to use as a default here

File: rockcraft.yaml [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+description: |
+  Snapcraft aims to make upstream developers' lives easier and as such is not
+  a single toolset, but instead is a collection of tools that enable the
+  natural workflow of an upstream to be extended with a simple release step
+  into Snappy.
+  This version of Snapcraft is able to build core22 snaps in destructive mode
+  only.
+license: GPL-3.0
+platforms:
+  amd64: #TODO
```
Comment (lengau) [inline rockcraft.yaml:14]: I gather this is for a later PR?

**Review by lengau — APPROVED** [review]
Looks nice and solid!

**Review by mr-cal — APPROVED** [review]
It wasn't easy without a readme :wink: but I packed a snap on amd64 and arm64 (I added the arm64 arch to the rockcraft.yaml).

**Pull request merged**